blogs/2022/09/15/dev-container-features.md

```diff
@@ -0,0 +1,139 @@
+
+The source code of a feature pack has two components: An install script (`install.sh`) and a configuration file (`devcontainer-feature.json`).
+
+```
++-- feature
+|    +-- devcontainer-feature.json
+|    +-- install.sh
+|    +-- (other files)
+
+```
+
+`install.sh`: The installation entry point script – it’s conceptually added as a layer of the image’s Dockerfile and executed during build-time. This entry point script can install tools such as languages (ruby) and tools (GitHub CLI).
+
+`devcontainer-feature.json`: This contains metadata about the feature pack, a set of options that can be passed to the install script during installation, as well as "pieces" of devcontainer.json that will be merged into the final dev container. For example, if any feature indicates `privileged`: true in their config, the entire dev container will be started with the `--privileged` flag.
+
+Feature packs can be authored in a variety of languages, the most straightforward being shell scripts. If a feature is authored in a different language, information about it should be included in the metadata so that users can make an informed choice about it.
+
+> **Note:** You should be sure that feature packs you release publicly check and install dependencies in addition to the feature. Furthermore, public features are very likely to be used from both arm64 or x86_64 machines - so be sure to adapt to this when possible.
+
+You may review `devcontainer-feature.json` properties [in the spec](https://containers.dev/implementors/features/#devcontainer-feature-json-properties), along with public examples in the [devcontainers/features](https://github.com/devcontainers/features) repo.
+
+Now that we’ve seen how to create a feature pack, how can I distribute it to others?
+
+## Distribution
+
+Feature packs are distributed as tarballs. The tarball contains the entire contents of the feature pack's sub-directory, including the `devcontainer-feature.json`, `install.sh`, and any other files in the directory.
+
+The [Open Container Initiative](https://opencontainers.org/), aka OCI, defines industry standards for containers and container resources. We treat feature packs as OCI Artifacts and use the concept of an [OCI Registry](https://containers.dev/implementors/features-distribution/#oci-registry) to distribute them.
+
+The [features template repository](https://github.com/devcontainers/feature-template) mentioned above includes a [GitHub Actions workflow](https://github.com/marketplace/actions/dev-container-publish) to automate the publishing process. It packages each feature into a tarball and publishes the assets as an OCI artifact to GHCR. Trigger the `release.yaml` workflow from the template repository by selecting it on the left of the repository’s Actions tab on GitHub. This will publish each feature to GHCR under the `<owner>/<repo>` namespace. A feature pack is only republished when the version property in its `devcontainer-feature.json` is updated.
```

Comment:
"It packages each feature into a tarball..." 
Unclear if each feature gets its own tarball or if multiple features are packaged (bundled) into a single tarball.
"This will publish each feature to GHCR..."
Should this be "feature pack"? Could be interpreted to mean "features" are pulled out separately and published.
Maybe "The GitHub Action will publish the feature pack to GHCR under...."

Reply:
Updated here and throughout to no longer use "feature pack" - let me know what you think.